api/resource.yaml

```diff
@@ -0,0 +1,2200 @@
+            type: string
+            format: uuid
+          description: Resource ID
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateActionRequest'
+            example:
+              name: "Create Reservation"
+              description: "Create a new reservation"
+              handler: "create"
+      responses:
+        "201":
+          description: Action created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Action'
+              example:
+                id: "be8f5i2i-9g0d-5fa4-d676-6f4eeg09ij7g"
+                name: "Create Reservation"
+                description: "Create a new reservation"
+                handler: "create"
+                resourceServerId: "3fa85f64-5717-4562-b3fc-2c963f66afa6"
+                resourceId: "7a4b1f8e-5c69-4b60-9232-2b0aaf65ef3c"
+        "400":
+          description: Bad request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-1001"
+                message: "Invalid request format"
+                description: "The request body is malformed or contains invalid data"
+        "404":
+          description: Resource or resource server not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              examples:
+                resource-not-found:
+                  summary: Resource not found
+                  value:
+                    code: "RES-1008"
+                    message: "Resource not found"
+                    description: "The resource with the specified id does not exist"
+                resource-server-not-found:
+                  summary: Resource server not found
+                  value:
+                    code: "RES-1003"
+                    message: "Resource server not found"
+                    description: "The resource server with the specified id does not exist"
+        "409":
+          description: Conflict
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-1004"
+                message: "Name conflict"
+                description: "An action with the same name already exists for this resource"
+        "500":
+          description: Internal server error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+              example:
+                code: "RES-5000"
+                message: "Internal server error"
+                description: "An unexpected error occurred while processing the request"
+
+  /resource-servers/{rsId}/resources/{resourceId}/actions/{id}:
```

Comment:
I'm not sure whether we should allow creating/ updating/ deleting actions at this level. Actions actually reside in Resource Server and has only associations with the Resources.

1. Creating a Action under Resource gave me the impression that I'm creating an Action there. But actually this will be created in the resource server and will be accessible to other Resources as well. However from a UI/UX point of view, this is a valid scenario. Hence fine to have create here.
2. IMO update and delete at this level should be only for updating/ deleting the association of the Resource to the Action (Can be merged to a new endpoint). Not to update/ delete the actual Action.

However agreed that we should have GET action at this level.

WDYT @darshanasbg @senthalan